dask/array/tests/test_rechunk.py

```diff
@@ -393,3 +393,79 @@ def test_dont_concatenate_single_chunks(shape, chunks):
     assert not any(funcname(task[0]).startswith('concat')
                    for task in dsk.values()
                    if dask.istask(task))
+
+
+def test_rechunk_unknown1():
+    dd = pytest.importorskip('dask.dataframe')
+    pd = pytest.importorskip('pandas')
+
+    x = dd.from_pandas(pd.DataFrame(np.random.randn(50, 10)), 2).values
+    # result = x.rechunk({1: 5})
+    result = x.rechunk((None, (5, 5)))
+    assert np.isnan(x.chunks[0]).all()
+    assert np.isnan(result.chunks[0]).all()
+    assert result.chunks[1] == (5, 5)
+
+
+def test_rechunk_unknown2():
+    import dask; dask.set_options(get=dask.get)  # noqa
```

Comment:
Why was this necessary?  If we need it then we might consider using a context manager instead to ensure cleanup.